### ISSUE TITLE
Warn (or error) when setuptools.build_meta:__legacy__ is explicitly specified

### ISSUE THREAD

**Issue by pganssle** [pypa/setuptools#1689]
Per the original discussion, `setuptools.build_meta:__legacy__` was intended *only* as a default and was not intended to be specified as the build backend. If we're going to go ahead with #1688, we'll need to be able to parse `pyproject.toml` anyway, so this could be a fairly simple way to start working with `pyproject.toml` and get started on #1688 without committing to support a whole new format.

I'd say it would be a good idea to get this done sooner rather than later, and preferably just make it an error directly, before too many people start using `setuptools.build_meta:__legacy__` directly.

**Comment by jaraco** [comment]
> it sorta breaks the separation between front-end and back-end in PEP 517

I agree - that seems like the wrong approach. I'd rather that pip (or other front-end tool) provide this protection.
